### DISCUSSION TITLE
Using UNNESTed values in WHERE condition

### DISCUSSION THREAD

**Discussion by venkyvb** [duckdb/duckdb#4846] [Q&A]
I have imported a parquet file into duckdb and am trying to query based on the UNNEST-ed values from a STRUCT field. I am able to run the following select statement it works fine (the STRUCT field is the `code.coding`) -
```
SELECT subject.reference,
    unnest(code.coding) c,
FROM observation o    
WHERE o.effective.datetime > '2017';
```
However, when I try to run something like this - 
```
SELECT subject.reference,
       AVG(value.quantity.value) avg_hdl
FROM observation o,
     UNNEST(o.code.coding) c
WHERE c.system = 'http://loinc.org' AND
      c.code = '2085-9' AND
      o.effective.datetime > '2017'
GROUP BY subject.reference
```
I am getting the following error -
```
(duckdb.BinderException) Binder Error: Referenced table "o" not found!
[SQL: SELECT subject.reference,
       AVG(value.quantity.value) avg_hdl
FROM observation o,
     UNNEST(o.code.coding) c
WHERE c.system = 'http://loinc.org/' AND
      c.code = '2085-9' AND
      o.effective.datetime > '2017'
GROUP BY subject.reference]
```
Wondering if there are examples of how such a query can be done ?

**Comment by Alex-Monahan** [comment]
Howdy! The reason is that unnest must only be used in the select clause, not the from.
https://duckdb.org/docs/sql/query_syntax/unnest

Could you provide an example dataset?

**Reply by venkyvb** [reply]
Thank you! 
You can use the sample data set - https://github.com/venkyvb/test-repo/blob/main/Observation_26d72cf3a52040b09eeb71b23ca8645c_00001.parquet

**Reply by venkyvb** [reply]
PS: I also tried to do the UNNEST in SELECT as follows which still results in an error -

```
SELECT subject.reference,
     UNNEST(o.code.coding) AS c
FROM observation o
WHERE c.system = 'http://loinc.org' AND
      c.code = '2085-9' AND
      o.effective.datetime > '2017';
```
Error
```
(duckdb.BinderException) Binder Error: Referenced table "c" not found!
Candidate tables: "o"
LINE 4: WHERE c.system = 'http://loinc.org/' AND
              ^
```

**Reply by Alex-Monahan** [reply]
Thanks for the data! I'll put a query together when I am at my computer.

In the meantime, the issue now is that we may need a subquery so that c is created before it is referenced in the where clause.

**Reply by venkyvb** [reply]
Thanks for that hint. I was able to get this working with a sub-query as following (not sure if this is the most optimized way of doing it) -
```
WITH coding as (SELECT UNNEST(observation.code.coding) c from observation)
SELECT subject.reference,
    AVG(value.quantity.value) avg_hdl
FROM observation o, coding
WHERE c.system = 'http://loinc.org' AND
    c.code = '2085-9' AND
      o.effective.datetime > '2017'
GROUP BY subject.reference;
```

**Reply by Alex-Monahan** [reply]
Nicely done! But you may need to join o and c (set up an equality condition in your where clause) to prevent duplication (which is hidden by your group by, but still skews the average)
